torch/csrc/autograd/python_variable.cpp

```diff
@@ -2148,8 +2150,12 @@ py::object torchDispatchFromTensorImpl(
   // NB: this may not be a python tensor if you got here from a mode!
   // TORCH_INTERNAL_ASSERT(isPythonTensor(self_t));
   append_overloaded_tensor(&overloaded_args, self_p.ptr());
-  auto args = py::reinterpret_steal<py::object>(PyTuple_New(1));
+  auto args = py::reinterpret_steal<py::object>(PyTuple_New(1 + (int)memory_format.has_value()));
```

Comment:
prefer static_cast over C-style cast, it is safer (it will refuse to implicitly reinterpret cast)